lib/elixir/lib/kernel.ex

```diff
@@ -780,8 +781,9 @@ defmodule Kernel do
 
       iex> tl([1, 2, 3, :go])
       [2, 3, :go]
+
   """
-  @spec tl(maybe_improper_list) :: maybe_improper_list
+  @spec tl(nonempty_maybe_improper_list) :: maybe_improper_list | term
```

Comment:
you mean: `nonempty_maybe_improper_list(any, any)` ?
if so, may I know what makes the difference.